Makefile

```diff
@@ -190,27 +190,26 @@ unit-sup: build-launcher-for-supervisor-tests
 .PHONY: build-launcher-for-supervisor-tests
 
 # Lints we need to work through and decide as a team whether to allow or fix
-UNEXAMINED_LINTS = clippy::cyclomatic_complexity \
-                   clippy::large_enum_variant \
-                   clippy::len_without_is_empty \
-                   clippy::module_inception \
-                   clippy::needless_pass_by_value \
-                   clippy::needless_return \
-                   clippy::new_ret_no_self \
-                   clippy::new_without_default \
-                   clippy::new_without_default_derive \
-                   clippy::question_mark \
-                   clippy::redundant_field_names \
-                   clippy::too_many_arguments \
-                   clippy::trivially_copy_pass_by_ref \
-                   clippy::wrong_self_convention \
-                   renamed_and_removed_lints
+UNEXAMINED_LINTS =
 
 # Lints we disagree with and choose to keep in our code with no warning
-ALLOWED_LINTS =
+ALLOWED_LINTS = clippy::module_inception \
+                clippy::needless_return \
+                clippy::new_ret_no_self \
+                clippy::new_without_default \
+                clippy::new_without_default_derive \
+                clippy::redundant_field_names
```

Comment:
https://rust-lang-nursery.github.io/rust-clippy/master/index.html#redundant_field_names

Personally, I'd rather change these as I think the shorthand version is more idiomatic, but I don't feel very strongly and I know that @raskchanky prefers the more explicit version, so I'll defer to what other people think.

[redundant_field_names.txt](https://github.com/habitat-sh/habitat/files/2813955/redundant_field_names.txt)

Reply:
Since we last spoke on this topic, I've become more comfortable with the shorthand version, so I'm happy to go that route.

Reply:
In light of @raskchanky's change of heart, I'm voting against this lint's current place in `ALLOWED_LINTS`. We should change these.

Reply:
I'm currently in the "explicit" camp, but I could be persuaded otherwise. I'm happy to try it out, particularly as it looks like it's becoming more idiomatic.

Reply:
I'm in the shorthand camp here. I think it's generally pretty obvious what's going on once you know the syntax.

Reply:
No strong opinion but I do like the look of the short version.